mmf/models/transformers/backends/huggingface.py

```diff
@@ -0,0 +1,218 @@
+            )
+            self.layer_norms.append(
+                torch.nn.LayerNorm(
+                    self.transformer_config.hidden_size, eps=layer_norm_eps
+                )
+            )
+            self.dropouts.append(nn.Dropout(hidden_dropout_prob))
+
+        self.token_type_embeddings = nn.Embedding(
+            len(self.model_config.modalities), self.transformer_config.hidden_size
+        )
+
+    def init_weights(self, transformer: Type[nn.Module]):
+        for idx, modality in enumerate(self.model_config.modalities):
+            if modality.type == "text":
+                self.token_embeddings[idx] = transformer.embeddings.word_embeddings
+                self.layer_norms[idx] = transformer.embeddings.LayerNorm
+
+            self.pos_embeddings[idx].weight = nn.Parameter(
+                deepcopy(transformer.embeddings.position_embeddings.weight.data),
+                requires_grad=True,
+            )
+
+        # Token Type or Segment Embeddings
+        if hasattr(transformer.embeddings, "token_type_embeddings"):
+            token_vocab_size = self.transformer_config.type_vocab_size
+            self.token_type_embeddings.weight.data[:token_vocab_size].copy_(
+                transformer.embeddings.token_type_embeddings.weight
+            )
+            for idx in range(token_vocab_size, len(self.model_config.modalities)):
+                self.token_type_embeddings.weight.data[idx].copy_(
+                    transformer.embeddings.token_type_embeddings.weight.data.mean(dim=0)
+                )
+                # Add random normal noise
+                self.token_type_embeddings.weight.data[idx] += torch.normal(
+                    self.model_config.token_noise_mean,
+                    self.model_config.token_noise_std,
+                    size=self.token_type_embeddings.weight.data[idx].size(),
+                )
+
+    def forward(
+        self,
+        input_ids: Dict[str, Tensor],
+        position_ids: Dict[str, Tensor],
+        segment_ids: Dict[str, Tensor],
+    ) -> Tensor:
+        list_embeddings = []
+        for idx, (token_emb, pos_emb, layer_norm, dropout) in enumerate(
+            zip(
+                self.token_embeddings,
+                self.pos_embeddings,
+                self.layer_norms,
+                self.dropouts,
+            )
+        ):
+            modality_name = self.modality_keys[idx]
+            total_embedding = token_emb(input_ids[modality_name])
+            if modality_name in position_ids:
+                total_embedding += pos_emb(position_ids[modality_name])
+
+            if modality_name in segment_ids:
+                total_embedding += self.token_type_embeddings(
+                    segment_ids[modality_name]
+                )
+
+            list_embeddings.append(dropout(layer_norm(total_embedding)))
+
+        return torch.cat(list_embeddings, dim=1)
+
+
+class HuggingfaceBackend(BaseTransformerBackend):
```

Comment:
Here now that they inherit from same class, you can add them to registry. This is similar to how each model inherits from BaseModel.

This is needed because internal files would require try except without registry pattern.